pkg/expr/expr_types_test.go

```diff
@@ -8,6 +8,149 @@ import (
 	"testing"
 )
 
+func Example_set_kv() {
+	var (
+		p = map[string]interface{}{
+			"kv":    &KV{},
+			"key":   "k1",
+			"value": Must(NewString("v11")),
+		}
+	)
+
+	eval(`set(kv, key, value)`, p)
+
+	// output:
+	// <nil>
+}
+
+func Example_merge_kv() {
+	var (
+		p = map[string]interface{}{
+			"kv": &KV{},
+			"foo": &KV{value: map[string]string{
+				"k1": "v1",
+			}},
+			"bar": &KV{value: map[string]string{
+				"k2": "v2",
+			}},
+		}
+	)
+
+	eval(`merge(kv, foo, bar)`, p)
+
+	// output:
+	// &{value:map[k1:v1 k2:v2]}
+}
+
+func Example_filter_kv() {
+	var (
+		p = map[string]interface{}{
+			"kv": &KV{value: map[string]string{
+				"k1": "v1",
+				"k2": "v2",
+			}},
+			"key1": "k1",
+			"key2": "k3",
+		}
+	)
+
+	eval(`filter(kv, key1, key2)`, p)
+
+	// output:
+	// &{value:map[k1:v1]}
+}
+
+func Example_omit_kv() {
+	var (
+		p = map[string]interface{}{
+			"kv": &KV{value: map[string]string{
+				"k1": "v1",
+				"k2": "v2",
+				"k3": "v3",
+			}},
+			"key1": "k1",
+			"key2": "k3",
+		}
+	)
+
+	eval(`omit(kv, key1, key2)`, p)
+
+	// output:
+	// &{value:map[k2:v2]}
+}
+
+func Example_set_kvv() {
+	var (
+		p = map[string]interface{}{
+			"kvv":   &KVV{},
+			"key":   "foo",
+			"value": Must(NewString("bar")),
+		}
+	)
+
+	eval(`set(kvv, key, value)`, p)
+
+	// output:
+	// <nil>
```

Comment:
Is this correct? Should test return <nil>? Or modified input?
So, is original input mutated here or does it stay unchanged after call? Would be very nice if we could prove that it's unchanged with the test?

Reply:
Sure